src/devices/Ili934x/Ili9341.cs

```diff
@@ -137,42 +158,59 @@ public Ili9341(SpiDevice spiDevice, int dataCommandPin, int resetPin, int backli
         /// <param name="y">The y co-ordinate of the point to start the rectangle at in pixels.</param>
         /// <param name="w">The width of the rectangle in pixels.</param>
         /// <param name="h">The height of the rectangle in pixels.</param>
-        public void FillRect(Color color, uint x, uint y, uint w, uint h)
+        /// <param name="doRefresh">True to immediately update the screen, false to only update the back buffer</param>
+        public void FillRect(Color color, int x, int y, int w, int h, bool doRefresh = true)
         {
             Span<byte> colourBytes = stackalloc byte[2]; // create a short span that holds the colour data to be sent to the display
-            Span<byte> displayBytes = stackalloc byte[(int)(w * h * 2)]; // span used to form the data to be written out to the SPI interface
 
             // set the colourbyte array to represent the fill colour
-            (colourBytes[0], colourBytes[1]) = Color565(color);
+            var c = Rgb565.FromRgba32(color);
 
             // set the pixels in the array representing the raw data to be sent to the display
             // to the fill color
-            for (int i = 0; i < w * h; i++)
+            for (int j = y; j < y + h; j++)
             {
-                displayBytes[i * 2 + 0] = colourBytes[0];
-                displayBytes[i * 2 + 1] = colourBytes[1];
+                for (int i = x; i < x + w; i++)
+                {
+                    _screenBuffer[i + j * ScreenWidth] = c;
+                }
             }
 
-            // specify a location for the rows and columns on the display where the data is to be written
-            SetWindow(x, y, x + w - 1, y + h - 1);
+            if (doRefresh)
```

Comment:
I'd suggest to create AutoRefresh property or similar and either do it everywhere or nowhere

Reply:
I think that would make the use more complex. ClearScreen is really the only command that should be used with an immediate effect. In all other cases, you need more than one drawing operation before you can publish the new screen content. I'm going to remove the doRefresh parameter from the public FillRectangle method.